### PR TITLE
Feature/overview

### PR DESCRIPTION
**Pull request by FilipGrebowski**

### CODE REVIEW

File: src/components/Core/Overview/OverviewPage.jsx [hunk 1]
```diff
@@ -19,35 +20,71 @@ class OverviewPage extends React.Component {
 
   async componentDidMount() {
     const music = MusicKit.getInstance();
-    let frequentlyPlayed = music.api.historyHeavyRotation();
+    // let frequentlyPlayed = music.api.historyHeavyRotation();
     let recentlyPlayed = music.api.recentPlayed();
 
-    frequentlyPlayed = await frequentlyPlayed;
+    // frequentlyPlayed = await frequentlyPlayed;
```
Comment (raphaelvigee) [inline src/components/Core/Overview/OverviewPage.jsx]: Useless

File: src/components/Core/Overview/OverviewPage.jsx [hunk 2]
```diff
@@ -19,35 +20,71 @@ class OverviewPage extends React.Component {
...
     recentlyPlayed = await recentlyPlayed;
 
     this.setState({
-      frequentlyPlayed,
+      // frequentlyPlayed,
```
Comment (raphaelvigee) [inline src/components/Core/Overview/OverviewPage.jsx]: Useless

File: src/components/Core/Overview/OverviewPage.jsx [hunk 3]
```diff
@@ -19,35 +20,71 @@ class OverviewPage extends React.Component {
...
   }
 
   render() {
     const {frequentlyPlayed, recentlyPlayed} = this.state;
+
+    if (!recentlyPlayed) {
+      return null;
+    }
+
+    console.log(recentlyPlayed);
```
Comment (raphaelvigee) [inline src/components/Core/Overview/OverviewPage.jsx]: No CL

File: src/components/Core/Overview/OverviewPage.jsx [hunk 4]
```diff
@@ -19,35 +20,71 @@ class OverviewPage extends React.Component {
...
+        <Loader/>
+        <h3>Recently played</h3>
+        <h4 className={classes.albumHeading}>Albums</h4>
+        <div className={classes.flexGrid}>
+          {
+            recentlyPlayed ?
+              recentlyPlayed.map((item, i) => {
+                switch (item.type) {
+                  case 'playlists':
+                    break;
```
Comment (raphaelvigee) [inline src/components/Core/Overview/OverviewPage.jsx]: Should return null

File: src/components/Core/Overview/OverviewPage.jsx [hunk 5]
```diff
@@ -19,35 +20,71 @@ class OverviewPage extends React.Component {
...
+          {
+            recentlyPlayed ?
+              recentlyPlayed.map((item, i) => {
+                switch (item.type) {
+                  case 'playlists':
+                    return (
+                      <AlbumItem key={i} album={item} size={120}/>
+                    );
+                  case 'albums':
+                    break;
```
Comment (raphaelvigee) [inline src/components/Core/Overview/OverviewPage.jsx]: Should return null

File: src/components/Core/common/SongList/SongListItem.jsx [hunk 6]
```diff
@@ -129,7 +129,7 @@ class SongListItem extends React.Component {
 
 
     return (
-      <div className={`${classes.song} ${this.isPlaying() ? 'playing' : ''}`}
+      <div className={`${!showAlbum ? classes.indexedSong : ''} ${classes.song} ${this.isPlaying() ? 'playing' : ''}`}
```
Comment (raphaelvigee) [inline src/components/Core/common/SongList/SongListItem.jsx]: Use `cx`

**Comment by BrychanOdlum** [comment]
Do not merge, we need to add more functionality first. Will help tonight.

File: src/components/Core/Overview/OverviewPage.jsx [hunk 7]
```diff
@@ -79,7 +77,7 @@ class OverviewPage extends React.Component {
                       <AlbumItem key={i} album={item} size={120}/>
                     );
                   case 'albums':
```
Comment (raphaelvigee) [inline src/components/Core/Overview/OverviewPage.jsx]:
```
default::
case 'album':
```

File: src/components/Core/common/SongList/SongListItem.jsx [hunk 8]
```diff
@@ -127,9 +127,8 @@ class SongListItem extends React.Component {
       </div>
     );
 
-
     return (
-      <div className={`${!showAlbum ? classes.indexedSong : ''} ${classes.song} ${this.isPlaying() ? 'playing' : ''}`}
+      <div className={cx({[classes.indexedSong]: !showAlbum}, classes.song, {[classes.playing]: this.isPlaying()})}
```
Comment (raphaelvigee) [inline src/components/Core/common/SongList/SongListItem.jsx]: Group the 2 classes objects

**Comment by mrjamesjarvis** [comment]
pls have a look - now mergeable

**Comment by BrychanOdlum** [comment]
Don't merge yet pls.

File: src/components/Core/ForYou/ForYouPage.jsx [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+
+    return (
+      <PageContent innerRef={this.ref}>
+        <PageTitle
+          title={"For You"}
+          context={"Apple Music"}
+        />
+        {recommendations.map((recList) => {
+          
+          const items = recList.relationships.contents == undefined ? recList.relationships.recommendations : recList.relationships.contents;
```
Comment (raphaelvigee) [inline src/components/Core/ForYou/ForYouPage.jsx]: `const items = recList.relationships.contents ? recList.relationships.contents : recList.relationships.recommendations;`

File: src/components/Core/ForYou/ForYouPage.jsx [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+    const music = MusicKit.getInstance();
+    let heavyRotation = music.api.historyHeavyRotation();
+    let recentlyPlayed = music.api.recentPlayed();
+    let recommendations = music.api.recommendations();
+    // ID 6: favourites, chill, new music mix
+    // ID 2: this days playlists
+    // ID 5: this days albums
+    // ID 3: artist spotlight playlists
+    // ID 9: new release albums
+    // There doesnt seem to be an order for this - so it may be user specific?
```
Comment (raphaelvigee) [inline src/components/Core/ForYou/ForYouPage.jsx]: Whatisdat ?

File: src/components/Core/ForYou/ForYouPage.scss [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+  > a {
+    margin: 0 0 20px;
+  }
+}
+
+.scrollWrapper {
+  width: 100%;
+  position: relative;
+  height: 200px;
+  &.groupedScroller {
```
Comment (raphaelvigee) [inline src/components/Core/ForYou/ForYouPage.scss]: \n

**Pull request merged**